pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -88,6 +88,8 @@ type AuthInfo struct {
 	ClientKeyData []byte `json:"client-key-data,omitempty"`
 	// Token is the bearer token for authentication to the kubernetes cluster.
 	Token string `json:"token,omitempty"`
+	// TokenFile is a pointer to a file that contains a bearer token (as described above).  If both Token and TokenFile are present, TokenFile takes precedence.
```

Comment:
precedence should follow the client cert filename versus data fields.  Data trumps filename, so token should trump tokenfile.